### PR TITLE
Consideration of imagePullPolicy on lifecycle hook execution

### LINKED ISSUE #12079: Lifecycle hooks not use imagePullPolicy
On pre / post ( RollingParams ) lifecycle, the ExecNewPod command don't use imagePullPolicy from baseContainer.

I think there are missing "ImagePullPolicy: baseContainer.ImagePullPolicy" at this line : https://github.com/openshift/origin/blob/master//pkg/deploy/strategy/support/lifecycle.go#L378

##### Version
OC 3.1

##### Steps To Reproduce

Push docker image with a tag ( not latest )
Launch deploy
Push docker image with the same tag ( not latest )
Relaunch deploy

##### Current Result

If image is already present host, oc execute pre / post command with old image 

##### Expected Result

Pull new version of image before execute pre / post command if imagePullPolicy is set to Always

##### Additional Information

Sample of DeploymentConfig used to reproduce :

{
  "kind": "DeploymentConfig",
  "apiVersion": "v1",
  "metadata": {
    "name": "frontend"
  },
  "spec": {
    "template": {
      "metadata": {
        "labels": {
          "name": "frontend"
        }
      },
      "spec": {
        "containers": [
          {
            "name": "helloworld",
            "image": "my-image:prod"
            "imagePullPolicy": "Always"
          }
        ]
      }
    }
    "replicas": 5,
    "selector": {
      "name": "frontend"
    },
    "strategy": {
      "type": "Rolling",
      "rollingParams": {
        "pre": {
          "failurePolicy": "Abort",
          "execNewPod": {
            "containerName": "helloworld"
          }
        }
      }
    }
  }
}

### PR DESCRIPTION
**Pull request by VanRoy**
For fixes #12079

### CODE REVIEW

File: pkg/deploy/strategy/support/lifecycle.go [hunk 1]
```diff
@@ -374,13 +374,14 @@ func makeHookPod(hook *deployapi.LifecycleHook, deployment *kapi.ReplicationCont
...
-					Resources:    resources,
-					VolumeMounts: volumeMounts,
+					Name:            HookContainerName,
+					Image:           baseContainer.Image,
+					ImagePullPolicy: baseContainer.ImagePullPolicy,
+					Command:         exec.Command,
+					WorkingDir:      baseContainer.WorkingDir,
+					Env:             mergedEnv,
+					Resources:       resources,
+					VolumeMounts:    volumeMounts,
```
Comment (pweil-) [inline pkg/deploy/strategy/support/lifecycle.go]: @kargakis should we be passing along other fields as well like SecurityContext settings?

Comment (kargakis) [inline pkg/deploy/strategy/support/lifecycle.go]: Sounds reasonable

Comment (VanRoy) [inline pkg/deploy/strategy/support/lifecycle.go]: Hi @kargakis , @pweil- , Do you want that i amend my PR with the other fields ? Can you list me all fields that you want to add ?

Comment (pweil-) [inline pkg/deploy/strategy/support/lifecycle.go]: @VanRoy created https://github.com/openshift/origin/issues/12103 to make sure we follow up on it.  No need to add it here.

**Comment by kargakis** [comment]
> @VanRoy created #12103 to make sure we follow up on it. No need to add it here.

Agreed. LGTM [merge]

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/10988 [merge]

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/12112 [merge]

**Comment by VanRoy** [comment]
Hi @kargakis , @pweil- , thanks for your rapid taking into account of my contribution.
Do you have any idea if this fix will be included in the next OCP 3.4 release ?

**Comment by pweil-** [comment]
@VanRoy I have created https://github.com/openshift/origin/issues/12121 for consideration in the 1.4 branch. 

@kargakis merge is assigned to you.  PTAL

**Comment by VanRoy** [comment]
@pweil- Awesome, thanks.

**Pull request merged**